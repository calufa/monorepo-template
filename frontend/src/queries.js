import { gql } from '@apollo/client'

export const GET_PROJECTS = gql`
  query GetProjects {
    projects {
      name
    }
  }
`
