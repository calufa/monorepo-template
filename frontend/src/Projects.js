import { useQuery } from '@apollo/client'
import React from 'react'
import styled from 'styled-components'
import { GET_PROJECTS } from './queries'

const Container = styled('div')`
	background: red;
	width: 200px;
	height: 200px;
`

export default function Projects() {
	const { loading, error, data } = useQuery(GET_PROJECTS)

	if (loading) return <p>Loading...</p>
	if (error) return <p>{JSON.stringify(error)}</p>

	return (
		<Container>
			{ data.projects[0].name }
		</Container>
	)
}
