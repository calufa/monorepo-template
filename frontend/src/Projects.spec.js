import { MockedProvider } from '@apollo/client/testing'
import 'jest-styled-components'
import React from 'react'
import { default as TestRenderer } from 'react-test-renderer'
import Projects from './Projects'
import { GET_PROJECTS } from './queries'

const mocks = [
  {
    request: {
      query: GET_PROJECTS,
    },
    result: {
      data: {
        projects: [
          {
            name: 'project-1',
          },
        ],
      },
    },
  },
]

it('it works', async () => {
  const render = TestRenderer.create(
    <MockedProvider addTypename={false} mocks={mocks}>
      <Projects />
    </MockedProvider>
  )
  await new Promise(e => setTimeout(e, 50))
  expect(render.toJSON()).toMatchSnapshot()
})
