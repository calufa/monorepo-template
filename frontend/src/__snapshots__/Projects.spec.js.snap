// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
.c0 {
  background: red;
  width: 200px;
  height: 200px;
}

<div
  className="c0"
>
  project-1
</div>
`;
